esmvalcore/_recipe.py

```diff
@@ -694,6 +703,70 @@ def _update_extract_shape(settings, config_user):
         check.extract_shape(settings['extract_shape'])
 
 
+def _parse_period(timerange):
+    start_year = None
+    end_year = None
+    if timerange.split('/')[0].startswith('P'):
+        try:
+            end_year = isodate.parse_datetime(timerange.split('/')[1]).year
+        except isodate.ISO8601Error:
+            end_year = isodate.parse_date(timerange.split('/')[1]).year
+        delta = int(isodate.parse_duration(timerange.split('/')[0]).years)
+        start_year = end_year - delta
+    elif timerange.split('/')[1].startswith('P'):
+        try:
+            start_year = isodate.parse_datetime(timerange.split('/')[0]).year
+        except isodate.ISO8601Error:
+            start_year = isodate.parse_date(timerange.split('/')[0]).year
+        delta = int(isodate.parse_duration(timerange.split('/')[1]).years)
+        end_year = start_year + delta
+    return start_year, end_year
+
+
+def _update_timerange(variable, settings, config_user):
+    if 'timerange' not in variable:
+        return
+
+    timerange = variable.get('timerange')
+    check.valid_time_selection(timerange)
+
+    step = 'clip_start_end_year'
+    if '*' in timerange:
+        (files, _, _) = _find_input_files(variable, config_user['rootpath'],
```

Comment:
Could you add a test case to tests/integration/data_finder.yml to demonstrate that this works?